pkg/terraform/exec/plugins/aws.go

```diff
@@ -3,13 +3,23 @@ package plugins
 import (
 	"github.com/hashicorp/terraform-plugin-sdk/plugin"
 	"github.com/terraform-providers/terraform-provider-aws/aws"
+
+	awstypes "github.com/openshift/installer/pkg/types/aws"
 )
 
 func init() {
-	exec := func() {
+	awsProvider := func() {
 		plugin.Serve(&plugin.ServeOpts{
 			ProviderFunc: aws.Provider,
 		})
 	}
-	KnownPlugins["terraform-provider-aws"] = exec
+	KnownPlugins["terraform-provider-aws"] = &TFPlugin{
+		Name:      awstypes.Name,
+		Exec:      awsProvider,
+		Resources: Resources(awsResources),
+	}
+}
+
+func awsResources() []string {
+	return []string{"vpc", "bootstrap", "cluster"}
```

Comment:
Is it possible to re-order this as `"cluster", "bootstrap"`, where the vpc part is done as part of the cluster target? I would expect that most platforms would do the bootstrap target after the cluster target.

Reply:
We had previously discussed this and agreed to create the vpc separately, so that is what was done here. If we were to change it back to the way it was and move it into "cluster", how would this even work? cluster requires the bootstrap resources to have been created, unless I am misunderstanding something here. I can say that this currently works "as-is" without any issues.

Reply:
That is not my recollection of where we left the previous discussion, but we can leave it as is for now.

Which bootstrap resources are required for the cluster resources? The bootstrap resources require cluster resources, specifically the vpc resources that you had to break out separately. I am not aware of any cluster resources that require bootstrap resources, however. But I admittedly have not dug into the tf file changes in depth yet.

Reply:
Maybe you can explain this to me on slack. If running the bootstrap sequence after cluster is possible, why do we even have a bootstrap sequence? My understanding of how nearly all openshift platforms are setup is this (simplified):

1. Create network resources
2. Setup bootstrap host
3. Setup cluster (control plane which I thought required the bootstrap host be up and configured). 

What am I missing here?

Reply:
The terraform is only creating the cloud resources, not directly running any OpenShift processes. In most cases, the control plane machines can be created before, after, or concurrently with the bootstrap machine. After the machines are created, the bootstrapping processes run on the bootstrap machine, and those processes allow the control plane machines to grab the ignition configs to ignite.

The notable exception to this is for Azure. Azure does not report its machines as created until after the machines ignite. So far Azure, the "cluster" setup would need to go after the "bootstrap" setup.

Let's leave the order as you have it for now. We can shuffle things around later (or not).

Reply:
Thank you for the explanation. I can certainly test out using a different order, however, I suspect it won't go as planned ;-) I think the original Terraform and code was written with assumptions on the order and probably requires more reworking.

Reply:
It looks like you were right. I changed the ordering to have bootstrap last and the cluster installed successfully!